src/Features/VisualBasic/Portable/ConvertIfToSwitch/VisualBasicConvertIfToSwitchCodeRefactoringProvider.Rewriting.vb

```diff
@@ -29,7 +29,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ConvertIfToSwitch
 
         Public Overrides Function AsSwitchSectionStatements(operation As IOperation) As IEnumerable(Of SyntaxNode)
             Dim node = operation.Syntax
-            Return If(node.IsStatementContainerNode(), node.GetStatements(), SpecializedCollections.SingletonEnumerable(node))
+            Dim statements = If(node.IsStatementContainerNode(), DirectCast(node.GetStatements(), IEnumerable(Of SyntaxNode)).ToArray(), {node})
```

Comment:
the cast to IEnumerable seems really strange.  why is it necessary?

Reply:
Oh, this is a leftover from when I was using .ToImmutableArray(). `ImmutableArray<StatementSyntax>` conflicted with `ImmutableArray<SyntaxNode>`, and `.ToImmutableArray(Of SyntaxNode)()` caused a VB error.